### PR TITLE
TST: add test for df comparing strings to numbers raises ValueError

### LINKED ISSUE #11565: Comparing Strings to Numbers: No ValueError
I noticed this quirk today: When you do a DataFrame-wide comparison (excluding `==`) using a number, it doesn't raise a ValueError (which you'd expect in Python 3); instead it always returns True.

``` python
>>> from pandas import DataFrame
>>> df = DataFrame(x: {'x': 'foo', 'y': 'bar', 'z': 'baz'} for x in ['a', 'b', 'c']})
>>> df
     a    b    c
x  foo  foo  foo
y  bar  bar  bar
z  baz  baz  baz
>>> df < 0
      a     b     c
x  True  True  True
y  True  True  True
z  True  True  True
>>> df > 0
      a     b     c
x  True  True  True
y  True  True  True
z  True  True  True
```

However, when you compare a Series of strings to a number, you get the expected ValueError:

``` python
>>> df.a < 0
TypeError: unorderable types: str() < int()
```

Is this a bug or a feature?

Python: 3.4.3
Pandas: 0.17.0
OS: Mac OSX 10.11

### PR DESCRIPTION
**Pull request by ganevgv**
- [x] closes #11565
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_operators.py [hunk 1]
```diff
@@ -530,6 +530,14 @@ def test_comp(func):
         test_comp(operator.ge)
         test_comp(operator.le)
 
+    def test_strings_to_numbers_comparisons_raises(self):
+        # GH 11565
+        df = DataFrame(
+            {x: {"x": "foo", "y": "bar", "z": "baz"} for x in ["a", "b", "c"]}
+        )
+        with pytest.raises(TypeError):
+            df > 0
```
Comment (mroeschke) [inline pandas/tests/frame/test_operators.py]: Can you `pytest.mark.parameterize` over all the operators? I think we have a fixture.

Comment (ganevgv) [reply pandas/tests/frame/test_operators.py]: Thanks, @mroeschke - now using parameterize to go over `["lt", "le", "gt", "ge", "eq", "ne"]`.

File: pandas/tests/frame/test_operators.py [hunk 2]
```diff
@@ -530,6 +530,34 @@ def test_comp(func):
         test_comp(operator.ge)
         test_comp(operator.le)
 
+    @pytest.mark.parametrize("op", ["lt", "le", "gt", "ge", "eq", "ne"])
```
Comment (mroeschke) [inline pandas/tests/frame/test_operators.py]: Can you use the `compare_operators_no_eq_ne` pytest fixture instead?

Comment (ganevgv) [reply pandas/tests/frame/test_operators.py]: Thanks, @mroeschke - that was neat! Now using `compare_operators_no_eq_ne`.

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ganevgv

**Pull request merged**